File: novalclaw-main/apps/omninova-android/app/src/main/java/com/omninova/phoneagent/data/ConversationModels.kt
package com.omninova.phoneagent.data

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * 与 `skills/phone-call-assistant/conversation_log_schema.json` 对齐。
 */
@Serializable
enum class ConversationChannel {
    @SerialName("voip_callkit") VOIP_CALLKIT,
    @SerialName("in_app_voice") IN_APP_VOICE,
    @SerialName("simulated") SIMULATED,
    @SerialName("cellular_telecom") CELLULAR_TELECOM,
    @SerialName("unknown") UNKNOWN,
}

@Serializable
data class ConversationTurn(
    val t: String,
    val role: String,
    val text: String,
    val confidence: Double? = null,
    @SerialName("raw_note") val rawNote: String? = null,
)

@Serializable
data class ConversationSessionFile(
    @SerialName("schema_version") val schemaVersion: String = "1.0",
    @SerialName("session_id") val sessionId: String,
    val channel: ConversationChannel,
    @SerialName("started_at_utc") val startedAtUtc: String,
    @SerialName("ended_at_utc") var endedAtUtc: String? = null,
    val locale: String? = null,
    val turns: MutableList<ConversationTurn> = mutableListOf(),
    var metadata: MutableMap<String, String>? = null,
)

File: novalclaw-main/apps/omninova-ios/OmniNovaPhoneAgent/Models/ConversationModels.swift
import Foundation

/// 与 `skills/phone-call-assistant/conversation_log_schema.json` 对齐。
enum ConversationChannel: String, Codable {
    case voipCallkit = "voip_callkit"
    case inAppVoice = "in_app_voice"
    case simulated = "simulated"
    case unknown = "unknown"
}

struct ConversationTurn: Codable, Identifiable {
    var id: String { "\(t)-\(role)" }
    let t: String
    let role: String
    let text: String
    var confidence: Double?
    var rawNote: String?

    enum CodingKeys: String, CodingKey {
        case t, role, text, confidence
        case rawNote = "raw_note"
    }
}

struct ConversationSessionFile: Codable, Identifiable {
    var id: String { sessionId }
    let schemaVersion: String
    let sessionId: String
    let channel: ConversationChannel
    let startedAtUtc: String
    var endedAtUtc: String?
    let locale: String?
    var turns: [ConversationTurn]
    var metadata: [String: String]?

    enum CodingKeys: String, CodingKey {
        case schemaVersion = "schema_version"
        case sessionId = "session_id"
        case channel
        case startedAtUtc = "started_at_utc"
        case endedAtUtc = "ended_at_utc"
        case locale
        case turns
        case metadata
    }

    static let schemaVersionValue = "1.0"
}
